pkgpanda/http/__init__.py

```diff
@@ -0,0 +1,229 @@
+"""Pkgpanda HTTP API"""
+
+import http.client
+import logging
+import os
+import sys
+
+from flask import Flask, current_app, jsonify, request
+
+from pkgpanda import Install, Repository, actions
+from pkgpanda.exceptions import (PackageConflict, PackageError,
+                                 PackageNotFound, ValidationError)
+
+
+empty_response = ('', http.client.NO_CONTENT)
+
+
+def package_listing_response(package_ids):
+    return jsonify(sorted(package_ids))
+
+
+def package_response(package_id, repository):
+    try:
+        package = repository.load(package_id)
+    except ValidationError:
+        response = (
+            error_response("Invalid package ID: {}".format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageNotFound:
+        response = (
+            error_response('Package {} not found.'.format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageError:
+        error_message = 'Unable to load package {}.'.format(package_id)
+        logging.exception(error_message)
+        response = (
+            error_response(error_message),
+            http.client.INTERNAL_SERVER_ERROR,
+        )
```

Comment:
Sure. Agree with your response. My intention with the first review comment was to avoid the repetition of multiple except clauses pattern that are repeated twice in this module. Introducing a new API based exception subclassing the parent could be done but I am with as it is if you feel that status quo is best.